src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -351,10 +351,37 @@ function VisEditor(
 
     $scope.isAddToDashMode = () => addToDashMode;
 
+    $scope.showSearchBarInline = () => {
+      // Show inline with menu, if it's only the timepicker \ autorefresh component
+      return ($scope.showQueryBarTimePicker() ||
+              $scope.showAutoRefreshOnly()) &&
+        !$scope.showQueryInput() &&
+        !$scope.showFilterBar();
+    };
```

Comment:
Something about the logic in this file is hard for me to grok. I think a lot of it is stemming from the fact that `showAutoRefreshOnly == !showQueryBarTimePIcker`

For example, in `showSearchBarInline`, the `return` statement starts with this conditional:
```js
// if I'm reading the code correctly, this will _always_ evaluate to `true`, so why include it?
return ($scope.showQueryBarTimePicker() || $scope.showAutoRefreshOnly()) && ...etc
```

Or: is it possible for `showQueryBar` to ever return `false`?

Maybe adding more details to your existing comments would help clarify?

Reply:
Holy guac, all visualizations have either time picker or autorefresh only => showQueryBar is always true. Love it!
I simplified the logic significantly and added a comment to the html file.